sdk/src/connections/target/rgbd_sensor.h

```diff
@@ -0,0 +1,111 @@
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef RGBD_SENSOR_H
+#define RGBD_SENSOR_H
+
+#include "addi9036_sensor.h"
+#include "aditof/depth_sensor_interface.h"
+#include "connections/target/v4l_buffer_access_interface.h"
+#include "rgb_sensor.h"
+
+#include <memory>
+
+namespace aditof {} // namespace aditof
+
+class RgbdSensor : public aditof::DepthSensorInterface {
+  public:
+    RgbdSensor(const std::shared_ptr<Addi9036Sensor> depthSensor,
+               const std::shared_ptr<RgbdSensor> rgbSensor);
+    ~RgbdSensor();
+
+  public: // implements DepthSensorInterface
+    virtual aditof::Status open() override;
+    virtual aditof::Status start() override;
+    virtual aditof::Status stop() override;
+    virtual aditof::Status
+    getAvailableFrameTypes(std::vector<aditof::FrameDetails> &types) override;
+    virtual aditof::Status
+    setFrameType(const aditof::FrameDetails &details) override;
+    virtual aditof::Status program(const uint8_t *firmware,
+                                   size_t size) override;
+    virtual aditof::Status getFrame(uint16_t *buffer,
+                                    aditof::BufferInfo *bufferInfo) override;
+    virtual aditof::Status readAfeRegisters(const uint16_t *address,
+                                            uint16_t *data,
+                                            size_t length) override;
+    virtual aditof::Status writeAfeRegisters(const uint16_t *address,
+                                             const uint16_t *data,
+                                             size_t length) override;
+    virtual aditof::Status
+    getDetails(aditof::SensorDetails &details) const override;
+    virtual aditof::Status getHandle(void **handle) override;
+    virtual aditof::Status getName(std::string &sensorName) const override;
+
+  public: // implements V4lBufferAccessInterface
+    // Methods that give a finer control than getFrame()
+    // And works if there is only one v4lbuffer (if many, then I don't know, maybe restructure this interface)
```

Comment:
There will be 2 v4lbuffers, one from Addio9036DepthSensor and one for RgbSensor. So here we need to changes things so that it works with multiple buffers.